src/planner/dart/ConfigurationToEndEffectorOffset.cpp

```diff
@@ -9,21 +10,79 @@ namespace dart {
 //==============================================================================
 ConfigurationToEndEffectorOffset::ConfigurationToEndEffectorOffset(
     statespace::dart::ConstMetaSkeletonStateSpacePtr stateSpace,
+    ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton,
     ::dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
-    const statespace::dart::MetaSkeletonStateSpace::State* startState,
     const Eigen::Vector3d& direction,
     const double signedDistance,
     constraint::ConstTestablePtr constraint)
-  : Problem(std::move(stateSpace), std::move(constraint))
+  : Problem(stateSpace, std::move(constraint))
+  , mMetaSkeletonStateSpace(stateSpace)
+  , mMetaSkeleton(std::move(metaSkeleton))
+  , mStartState(std::move(mMetaSkeletonStateSpace->createState()))
   , mEndEffectorBodyNode(std::move(endEffectorBodyNode))
-  , mStartState(startState)
-  , mDirection(direction.normalized())
+  , mDirection(direction)
   , mDistance(signedDistance)
 {
-  if (direction.squaredNorm() == 0)
+  if (mDirection.get().squaredNorm() == 0)
```

Comment:
Shouldn't this be `<= 1e-3` or something that's not exactly zero?

Reply:
I think this is only here because normalizing a zero vector throws an error. I don't think it's wrong to pass in a small-magnitude vector and use that resulting direction, so I think comparing exactly to zero is fine. However, I think it might be better to use `.isZero()` rather than `.squaredNorm() == 0`.